kolena/_experimental/object_detection/evaluator.py

```diff
@@ -0,0 +1,343 @@
+        baseline_pr_plot = baseline_pr_plot.curves[0] if baseline_pr_plot is not None else None
+        average_precision = (
+            compute_average_precision(baseline_pr_plot.y, baseline_pr_plot.x)
+            if baseline_pr_plot is not None and precision > 0
+            else 0.0
+        )
+
+        # compute nested metrics per class
+        labels = {gt.label for _, gts, _ in inferences for gt in gts.bboxes}
+        per_class_metrics: List[ClassMetricsPerTestCase] = []
+        for label in labels:
+            metrics_per_class = self.compute_aggregate_label_metrics(all_bbox_matches, label, thresholds)
+            per_class_metrics.append(metrics_per_class)
+
+        return TestCaseMetrics(
+            TestSamples=len(inferences),
+            PerClass=per_class_metrics,
+            Objects=tp_count + fn_count,
+            Inferences=tp_count + fp_count,
+            TP=tp_count,
+            FN=fn_count,
+            FP=fp_count,
+            TPR=sum(1 for im in metrics if im.has_TP) / len(metrics) if len(metrics) > 0 else 0.0,
+            FPR=sum(1 for im in metrics if im.has_FP) / len(metrics) if len(metrics) > 0 else 0.0,
+            Precision=precision,
+            Recall=recall,
+            F1=2 * precision * recall / (precision + recall) if (precision + recall) > 0 else 0.0,
+            AP=average_precision,
+        )
+
+    def compute_test_case_plots(
+        self,
+        test_case: TestCase,
+        inferences: List[Tuple[TestSample, GroundTruth, Inference]],
+        metrics: List[TestSampleMetrics],
+        configuration: Optional[ThresholdConfiguration] = None,
+    ) -> Optional[List[Plot]]:
+        assert configuration is not None, "must specify configuration"
+        thresholds = self.get_confidence_thresholds(configuration)
+        labels = {gt.label for _, gts, _ in inferences for gt in gts.bboxes}
+        all_bbox_matches = [
+            match_inferences_multiclass(
+                ground_truth.bboxes,
+                [
+                    inf
+                    for inf in inference.bboxes
+                    if inf.label in labels
+                    and inf.score >= max(configuration.min_confidence_score, thresholds[inf.label])
+                ],
+                ignored_ground_truths=ground_truth.ignored_bboxes,
+                mode="pascal",
+                iou_threshold=configuration.iou_threshold,
+            )
+            for _, ground_truth, inference in inferences
+        ]
+
+        plots: Optional[List[Plot]] = []
+        plots.extend(
+            filter(
+                None,
+                [
+                    compute_f1_plot_multiclass(all_bbox_matches),
+                    compute_f1_plot(all_bbox_matches),
+                    compute_pr_plot_multiclass(all_bbox_matches),
+                    compute_pr_plot(all_bbox_matches),
+                    compute_confusion_matrix_plot(all_bbox_matches),
+                ],
+            ),
```

Comment:
I'm confused as to what's going on here. We're plotting both `f1_plot` and `f1_plot_multiclass`? Neither have optional return types.

Reply:
Just wanted to temporarily inspect test case plots separately. Will fix signatures/logic!